### ISSUE TITLE
DefaultAzureCredential not behaving according to the docs

### ISSUE THREAD

**Issue by riedgar-ms** [Azure/azure-sdk-for-python#26271]
- **azure.identity**: 
- **1.7.0**: 
- **Linux 5.4.0-1083-azure # 87~18.04.1-Ubuntu SMP**:
- **3.8.5**: 

**Describe the bug**

In a VM deployed within AzureML, I'm trying to run a Python program from the command line, and authenticate using `DefaultAzureCredential()`. The [docs for this class](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) state that _"When an access token is needed, it requests one using these identities in turn, stopping when one provides a token"_ and provides a list of six different authentication methods. However, it fails after two and then gives up. This is plainly not the behaviour which the linked page describes.

**To Reproduce**

```python
Python 3.8.5 (default, Sep  4 2020, 07:30:14) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from azure.identity import DefaultAzureCredential
>>> DefaultAzureCredential().get_token()
DefaultAzureCredential failed to retrieve a token from the included credentials.
Attempted credentials:
        EnvironmentCredential: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.
Visit https://aka.ms/azsdk/python/identity/environmentcredential/troubleshoot to troubleshoot.this issue.
        ManagedIdentityCredential: "get_token" requires at least one scope
To mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azsdk/python/identity/defaultazurecredential/troubleshoot.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/anaconda/envs/azureml_py38/lib/python3.8/site-packages/azure/identity/_credentials/default.py", line 172, in get_token
    return super(DefaultAzureCredential, self).get_token(*scopes, **kwargs)
  File "/anaconda/envs/azureml_py38/lib/python3.8/site-packages/azure/identity/_credentials/chained.py", line 108, in get_token
    raise ClientAuthenticationError(message=message)
azure.core.exceptions.ClientAuthenticationError: DefaultAzureCredential failed to retrieve a token from the included credentials.
Attempted credentials:
        EnvironmentCredential: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.
Visit https://aka.ms/azsdk/python/identity/environmentcredential/troubleshoot to troubleshoot.this issue.
        ManagedIdentityCredential: "get_token" requires at least one scope
To mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azsdk/python/identity/defaultazurecredential/troubleshoot.
```

However:
```python
DefaultAzureCredential(exclude_managed_identity_credential=True)
```
is successful (it retrieves the CLI credential, which is the penultimate one listed in the doc.

**Expected behavior**

The constructor should behave as described - each method should be tried in turn and if it fails, go onto the next. Having to fiddle around with constructor arguments to `DefaultAzureCredential()` dependent on which machine is in use defeats the point.

**Screenshots**

**Additional context**

**Comment by riedgar-ms** [comment]
Note, having just created a fresh conda environment and installed all new packages, this does appear to be fixed in 1.10.0 at least.

**Comment by kristapratico** [comment]
Thanks for your issue @riedgar-ms. @xiangyan99 to confirm the behavior.
